### PR TITLE
Make exception messages in Compilation.cs localizable

### PR DESCRIPTION
**Pull request by jcouv**
Some exception messages were hardcoded, rather than pulling the string from localizable resource files.

Fixes https://github.com/dotnet/roslyn/issues/14302

@dotnet/roslyn-compiler @CyrusNajmabadi  for review. I'd like to include this in RC so that the strings can be localized by RTM.

### CODE REVIEW

File: src/Compilers/Core/Portable/CodeAnalysisResources.resx [hunk 1]
```diff
@@ -522,6 +522,21 @@
     <value>Exception occurred with following context:
 {0}</value>
   </data>
+  <data name="AnonymousTypeMemberAndNamesCountMismatch2" xml:space="preserve">
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/CodeAnalysisResources.resx]: I recommend following the pattern the IDE uses.  IN this case, you'd have "_0_and_1_must_have_the_same_length" as the name of the string.  This makes things *super* easy to find when you're trying to go back from a message in an error list to the actual loc string. 

Your call though.

File: src/Compilers/Core/Portable/CodeAnalysisResources.resx [hunk 2]
```diff
@@ -522,6 +522,21 @@
...
+    <value>{0} and {1} must have the same length.</value>
+  </data>
+  <data name="AnonymousTypeArgumentCountMismatch2" xml:space="preserve">
+    <value>{0} must either be 'default' or have the same length as {1}.</value>
+  </data>
+  <data name="InconsistentSyntaxTreeFeature" xml:space="preserve">
+    <value>inconsistent syntax tree features</value>
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/CodeAnalysisResources.resx]: Why lowercase?

**Comment by CyrusNajmabadi** [comment]
👍

File: src/Compilers/Core/Portable/CodeAnalysisResources.resx (hunk 1 above)
Comment (jcouv) [inline src/Compilers/Core/Portable/CodeAnalysisResources.resx]: There was pushback from the compiler team on that naming convention in a previous PR.

**Comment by VSadov** [comment]
LGTM

**Comment by jcouv** [comment]
@MattGertz Please approve for RC. This is a minor fix to make error messages localizable.

**Comment by MattGertz** [comment]
Absolutely.

**Comment by OmarTawfik** [comment]
@jcouv does this close #14313 as well?

**Comment by jcouv** [comment]
Not sure. I only scanned through Compilation.cs, which had known issues. This makes me wonder if we should do a broader scan.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Pull request merged**